airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -0,0 +1,225 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from abc import ABC, abstractmethod
+from functools import partial
+from typing import Any, Callable, Iterator, Mapping, MutableMapping, Sequence
+
+import requests
+from requests import HTTPError
+
+
+class FreshdeskError(HTTPError):
+    """
+    Base error class.
+    Subclassing HTTPError to avoid breaking existing code that expects only HTTPErrors.
+    """
+
+
+class FreshdeskBadRequest(FreshdeskError):
+    """Most 40X and 501 status codes"""
+
+
+class FreshdeskUnauthorized(FreshdeskError):
+    """401 Unauthorized"""
+
+
+class FreshdeskAccessDenied(FreshdeskError):
+    """403 Forbidden"""
+
+
+class FreshdeskNotFound(FreshdeskError):
+    """404"""
+
+
+class FreshdeskRateLimited(FreshdeskError):
+    """429 Rate Limit Reached"""
+
+
+class FreshdeskServerError(FreshdeskError):
+    """50X errors"""
+
+
+class API:
+    def __init__(self, domain: str, api_key: str, verify: bool = True, proxies: MutableMapping[str, str] = None):
+        """Basic HTTP interface to read from endpoints"""
+        self._api_prefix = f"https://{domain.rstrip('/')}/api/v2/"
+        self._session = requests.Session()
+        self._session.auth = (api_key, "unused_with_api_key")
+        self._session.verify = verify
+        self._session.proxies = proxies
+        self._session.headers = {"Content-Type": "application/json"}
+
+        if domain.find("freshdesk.com") < 0:
+            raise AttributeError("Freshdesk v2 API works only via Freshdesk domains and not via custom CNAMEs")
+
+    @staticmethod
+    def _parse_and_handle_errors(req):
```

Comment:
when would the API not return a JSON response? Doesn't a non-JSON response mean there was a problem in the request?

Reply:
I think you are right, I will remove try-except, to not ignore errors silently
this part also originated from freshdesk-python

Reply:
because we handle all kind of HTTP responses, there could response from web server (status=500, 403 or 404, etc)